packages/react-scripts/bin/react-scripts.js

```diff
@@ -24,7 +24,7 @@ const scriptIndex = args.findIndex(
 const script = scriptIndex === -1 ? args[0] : args[scriptIndex];
 const nodeArgs = scriptIndex > 0 ? args.slice(0, scriptIndex) : [];
 
-if (['build', 'eject', 'start', 'test'].includes(script)) {
+if (['build', 'eject', 'start', 'test', 'build-component'].includes(script)) {
```

Comment:
it looks like you're on the right path here. i don't know all the places you have to declare this new `build-component` script exists, but we could probably just grep for `build` or whatever to find out.